src/librustc/ty/error.rs

```diff
@@ -178,6 +165,11 @@ impl<'tcx> fmt::Display for TypeError<'tcx> {
                        values.expected.ty,
                        values.found.ty)
             }
+            ExistentialMismatch(ref values) => {
+                write!(f, "conflicting existentials: `{}` and `{}`",
```

Comment:
Use `report_maybe_different`. Also, don't leak the term "existential" (or "dynamic") to user.